platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,346 @@
+| TSS-008        | The ledger signature SHOULD be verifiable by an EVM smart contract without excessive cost or time.                           |
+| TSS-009        | The TSS implementation SHOULD be able to switch elliptic curves.                                                             |
+| TSS-010        | The TSS algorithm SHOULD be able to model consensus weight with high precision.                                              |
+
+### Design Decisions
+
+#### New Address Book Life-Cycle
+
+This design proposal will deliver after Dynamic Address Book  (DAB) Phase 2 and before DAB Phase 3. In Phase 2 the
+address book is updated on software upgrades. In Phase 3 the address book is updated dynamically without restarting
+the node. Since the TSS effort requires keying a roster with enough `TssMessages` to generate ledger signatures, a
+modified life-cycle is needed. Some of the dynamic work in Phase 3 will be needed. This work is being encapsulated
+in the TSS-Roster proposal, which will need to be completed before the TSS-Ledger-Id proposal can be fully implemented.
+
+The relevant Address Book life-cycle changes are the following:
+
+1. The platform only receives a Roster, a subset of the Address Book.
+2. The application address book has 3 states: `Active` (AAB), `Candidate` (CAB), and `Future` (FAB).
+3. The FAB is updated by every HAPI transaction that is already received in DAB Phase 2.
+4. The CAB is a snapshot of the FAB when the APP is ready to initiate an address book rotation.
+5. The AAB is replaced by the CAB on a software upgrade if the consensus roster derived from the CAB has enough key
+   material to generate ledger signatures.
+6. The CAB is not adopted and the previous AAB remains the existing AAB after a software upgrade if the CAB does not
+   have enough key material to generate ledger signatures.
+
+Prior to restart for upgrade,  `candidate` consensus roster will be set in the platform state by the app in such a way
+as to mark it clearly as the next consensus roster. This methodology replaces the previous methodology in DAB Phase
+2, which was to write a new `config.txt` to disk. After the `TSS-Roster` proposal has been implemented, the only
+need for a file on disk is the genesis address book at the start of a new network.
+
+Once the correct address book and consensus roster life-cycle is in place through the `TSS-Roster` proposal, this
+proposal only introduces the logistics of generating the key material and the logic for deciding to adopt the CAB in
+replacement of the AAB.
+
+#### TSS Algorithm
+
+The threshold signature scheme chosen for our implementation is detailed in https://eprint.iacr.org/2021/339. This
+scheme is able to meet the core requirements listed above. The rest of this proposal details the design of how to
+implement this scheme in the consensus nodes.
+
+The Groth21 algorithm uses Shamir Secret Sharing to hide the private key of the ledger and distribute shares of it
```

Comment:
I have no idea what Shamir Secret Sharing is. A very short description like "Shamir Secret Sharing is an algorithm that is used in many TSS algorithms for key hiding" would be enough.

Reply:
I will ask the reader to refer to the `TSS-Library` effort for greater details.

Reply:
I'll skip adding this link: https://letmegooglethat.com/?q=Shamir+Secret+Sharing